app/assets/javascripts/ajax_ops.js

```diff
@@ -92,5 +108,147 @@ function handle_ajax(event) {
+        }).then((response) => {
+          if (response.status === 200) {
+          response.json().then((data) => {
+            resultsDiv.innerHTML = '';
+            let parag = document.createElement('P');
+            parag.textContent = JSON.stringify(data);
+            resultsDiv.appendChild(parag);
+          });
+        } else {
+          response.json().then((data) => {
+            alert(`Return code ${response.status} ${response.statusText} ${JSON.stringify(data)}`);
+          }).catch((error) => {
+            console.log(error);
+            alert(error);
+          });
+        }
+        });
+    }
+    // List user's facts
+    else if (event.target === listUserFactsButton) {
+      fetch(`${users_path}/${factsUserID.value}/facts`,
+      {
+        method: 'GET',
+        headers: {'Content-Type': 'application/json'}
+      }).then((response) => {
+        if (response.status === 200) {
+        response.json().then((data) => {
+          resultsDiv.innerHTML = '';
+          let parag = document.createElement('P');
+          parag.textContent = JSON.stringify(data);
+          resultsDiv.appendChild(parag);
+        });
+      } else {
+        response.json().then((data) => {
+          alert(`Return code ${response.status} ${response.statusText} ${JSON.stringify(data)}`);
+        }).catch((error) => {
+          console.log(error);
+          alert(error);
+        });
+      }
+      });
+    }
+    // Create a fact
+    else if (event.target === createFactButton) {
+      var dataObject = {
+        fact_text: fact.value,
+        likes: likes.value
+      }
+      if (dataObject.fact === "") {  // blank fact is not supported
+        delete dataObject.fact;
+      }
+      if (dataObject.likes === "") { // blank likes is not supported
+        delete dataObject.likes;
+      }
+      fetch(`${users_path}/${createFactUserID.value}/facts`,
+      {
+        method: 'POST',
+        headers: {'Content-Type': 'application/json'},
+        body: JSON.stringify(dataObject)
+      }).then((response) => {
+        if (response.status === 201) {
+          response.json().then((data) => {
+            resultsDiv.innerHTML = '';
+            let parag = document.createElement('P');
+            parag.textContent = JSON.stringify(data);
+            resultsDiv.appendChild(parag);
+          });
+        } else {
+          response.json().then((data) => {
+            alert(`Return code ${response.status} ${response.statusText} ${JSON.stringify(data)}`);
+          }).catch((error) => {
+            console.log(error);
+            alert(error);
+          });
+        }
+      });
+    }
+    // Update a fact
+    else if (event.target === updateFactButton) {
+      var dataObject = {
+        fact_text: updateFact.value,
+        likes: updateLikes.value
+      }
+      if (dataObject.updateFact === "") {  // blank fact is not supported
+        delete dataObject.updateFact;
```

Comment:
Same here, it's called `fact_text` above so that is how it should be called here. `dataObject.fact_text`